src/test/scala/nl/biopet/tools/bamstats/ScatterAndMergeTest.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2014 Biopet
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package nl.biopet.tools.bamstats
+
+import java.io.File
+
+import nl.biopet.test.BiopetTest
+import nl.biopet.tools.bamstats.schema.BamstatsRoot
+import nl.biopet.utils.ngs.intervals.BedRecordList
+import org.testng.annotations.{AfterClass, Test}
+import nl.biopet.tools.scatterregions.ScatterRegions.scatterRegions
+import scala.collection.mutable.ListBuffer
+
+class ScatterAndMergeTest extends BiopetTest {
+
+  val testDirs: ListBuffer[File] = ListBuffer()
+  def tempDir(prefix: String, suffix: String): File = {
+    val dir = File.createTempFile(prefix, suffix)
+    dir.delete()
+    dir.mkdirs()
+    testDirs += dir
+    dir
+  }
+
+  def mkdir(file: File): File = {
+    file.delete()
+    file.mkdirs()
+    file
+  }
+
+  @AfterClass
+  def deleteTestDirs(): Unit = {
+    testDirs.toList.foreach { dir =>
+      dir.delete()
```

Comment:
delete on a non-empty did does not work. You should use a recusive method. I think common.io (apache) should have a method like this.